### PR TITLE
[feature][vx-brush] add Brush component

### LINKED ISSUE #14: [brush] add <BrushX /> and <BrushY /> components

### PR DESCRIPTION
**Pull request by geekplux**
Closes: #556 

#### :rocket: Enhancements

- add Brush component writing by typescript; fix #14 

image

Inspired and migrated from [data-ui](https://github.com/williaster/data-ui/tree/master/packages/xy-chart/src/utils/brush)
Please check and review.

### CODE REVIEW

**Comment by geekplux** [comment]
Now the movement of BrushSelection in minimap is not smooth, I guess it is caused by the way update the main view [here](https://github.com/hshoff/vx/pull/553/files#diff-5040c402f5701145024fe41db2b452e4R108-R117). I don't know if we change it to a new way e.g. using zoom or rescale might be better.

**Comment by williaster** [comment]
Thanks @geekplux for the migration/addition! I will take a look at the PR ASAP in the next day or two 🎉

**Comment by hshoff** [comment]
Woo! Thanks for the PR. Haven't had a chance to do an full review but a quick read through looks good. 

Looks like this PR needs to update tests for `vx-vx` and `vx-brush` and that should make test CI happy. a few minor fixes for the lint check to pass as well. Not blockers as these are relatively minor things.

File: packages/vx-brush/src/BaseBrush.tsx [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+        {/* overlay */}
+        <Drag
+          width={stageWidth}
+          height={stageHeight}
+          resetOnStart
+          onDragStart={this.handleDragStart}
+          onDragMove={this.handleDragMove}
+          onDragEnd={this.handleDragEnd}
+        >
+          {(draw: any) => (
```
[the remark this answers is not in the document]
Comment (geekplux) [reply packages/vx-brush/src/BaseBrush.tsx]: Great

**Comment by geekplux** [comment]
I saw the [lint test](https://travis-ci.org/hshoff/vx/jobs/613835077?utm_medium=notification&utm_source=github_status) is still failed, but there is no detail info to figure out which lines caused these errors.

**Comment by hshoff** [comment]
```js
TypeError: Cannot set property 'hasAwait' of null
Occurred while linting ~/vx/packages/vx-responsive/test/ScaleSVG.test.tsx:12
     at Object.AwaitExpression (~/vx/node_modules/eslint/lib/rules/require-await.js:92:36)
     at ReturnStatement (~/vx/node_modules/@typescript-eslint/eslint-plugin/dist/rules/require-await.js:68:27)
```
https://github.com/hshoff/vx/blob/master/packages/vx-responsive/test/ScaleSVG.test.tsx

Can probably just disable the eslint rule for now:

```js
  test('it should expose its ref via an innerRef prop', () => {
    // eslint-disable-next-line require-await
    return new Promise(done => {
```

**Comment by geekplux** [comment]
seems no `--clean` and `--extensions` in babel options which caused all checks failed
https://babeljs.io/docs/en/options

**Comment by williaster** [comment]
@geekplux thanks again for your work on this! I am going to try and wrap up the TS project over the next week and experiment with an alpha release which will include this. Will ping you when it's out! 🙌

**Comment by williaster** [comment]
Test coverage went down which is why the build is 🔴 . Going to merge and hope we can circle back on tests.

**Comment by geekplux** [comment]
No problem, feel free to contact me if any issues. Merry Christmas!

**Comment by williaster** [comment]
@geekplux playing with this locally I don't get the brush selection. I should have time to debug today but if you have thoughts lmk 👍 

image

**Comment by williaster** [comment]
I think I figured it out. Will post a PR soon with a couple TS changes as well and tag you 👍

**Comment by geekplux** [comment]
Thanks a lot. It's LGTM on my local windows PC and I didn't test it on other machines. It might be a CSS error if only selection area is hidden. Anyway, sounds good the solution you have found.

**Comment by jskrt** [comment]
Is there any documentation on how to use this?

**Pull request merged**